### PR TITLE
UIPER-104: leverage cookie based authentication in all api requests

### PR DESCRIPTION
**Pull request by elsenhans**
https://issues.folio.org/browse/UIPER-104

### CODE REVIEW

File: src/util/downloadCSV.js [hunk 1]
```diff
@@ -8,8 +10,9 @@ function downloadCSV(url, stripes, params, mocks) {
   return fetch(`${cleanedURL}&csv=true`, {
     headers: {
       'X-Okapi-Tenant': stripes.okapi.tenant,
-      'X-Okapi-Token': stripes.okapi.token,
-    }
+      ...getLegacyTokenHeader(stripes.okapi),
+    },
+    credentials: 'include',
```
Comment (alb3rtino) [inline src/util/downloadCSV.js:15]:
```suggestion
      ...(stripes.okapi.token ? { 'X-Okapi-Token': stripes.okapi.token } : {}),
    },
    credentials: 'include',
```

**Review by alb3rtino — CHANGES_REQUESTED** [review]
Looks good! Since there is only a single use of the `X-Okapi-Token` header throughout the codebase, I suggest inlining the header. This keeps the code simple and removes the need for the `getLegacyTokenHeader.js` file, reducing unnecessary abstraction.

**Comment by MikeTaylor** [comment]
`getLegacyTokenHeader` should be provided by stripes-core.

Then using it as in the original version of this PR becomes a recognizable idiom.

**Comment by alb3rtino** [comment]
> `getLegacyTokenHeader` should be provided by stripes-core.
> 
> Then using it as in the original version of this PR becomes a recognizable idiom.

I guess using `okapiKy` or `stripes-connect` would be.

File: src/util/downloadCSV.js (hunk 1 above)
Comment (elsenhans) [reply src/util/downloadCSV.js]: Thanks @alb3rtino !
Outsourcing of `getLegacyTokenHeader` is build back

**Pull request merged**